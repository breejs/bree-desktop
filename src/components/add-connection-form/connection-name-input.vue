<script setup>
import { onMounted, ref, watchEffect, computed } from 'vue';
import isSANB from 'is-string-and-not-blank';
import useDebounceInput from '@/composables/debounce-input';

const props = defineProps({
  connectionNames: {
    type: Array,
    default: () => []
  },
  value: {
    type: String,
    default: 'localhost'
  }
});

const emit = defineEmits(['update:value']);

const input = ref(null);

const { inputValue, valid, classes } = useDebounceInput(
  props.value,
  'localhost',
  input,
  emit,
  (value) => isSANB(value) && !props.connectionNames?.includes(value)
);

defineExpose({ valid });
</script>

<template lang="pug">
.form-floating
  input#connection-name-input.form-control(
    name='name',
    type='text',
    placeholder='',
    required,
    v-model='inputValue',
    :class='classes',
    ref='input'
  )
  label(for='connection-name-input')
    | Name*
  .invalid-feedback
    | Enter a valid name that is unique
</template>
