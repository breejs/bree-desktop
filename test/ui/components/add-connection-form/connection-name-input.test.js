import { storyTests } from '../../_story-tests';

import * as stories from '@/components/add-connection-form/connection-name-input.stories';

storyTests(stories);
