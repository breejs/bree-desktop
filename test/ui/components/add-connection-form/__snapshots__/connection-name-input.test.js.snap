// Vitest Snapshot v1

exports[`story snapshots > Default 1`] = `
"<form novalidate=\\"\\">
  <div class=\\"form-floating\\"><input class=\\"form-control\\" id=\\"connection-name-input\\" name=\\"name\\" type=\\"text\\" placeholder=\\"\\" required=\\"\\"><label for=\\"connection-name-input\\">Name*</label>
    <div class=\\"invalid-feedback\\">Enter a valid name that is unique</div>
  </div>
</form>"
`;

exports[`story snapshots > Invalid 1`] = `
"<form novalidate=\\"\\">
  <div class=\\"form-floating\\"><input class=\\"form-control\\" id=\\"connection-name-input\\" name=\\"name\\" type=\\"text\\" placeholder=\\"\\" required=\\"\\"><label for=\\"connection-name-input\\">Name*</label>
    <div class=\\"invalid-feedback\\">Enter a valid name that is unique</div>
  </div>
</form>"
`;

exports[`story snapshots > Valid 1`] = `
"<form novalidate=\\"\\">
  <div class=\\"form-floating\\"><input class=\\"form-control\\" id=\\"connection-name-input\\" name=\\"name\\" type=\\"text\\" placeholder=\\"\\" required=\\"\\"><label for=\\"connection-name-input\\">Name*</label>
    <div class=\\"invalid-feedback\\">Enter a valid name that is unique</div>
  </div>
</form>"
`;
